### DISCUSSION TITLE
Strongswan IPSEC and ANYCAST

### DISCUSSION THREAD

**Discussion by p088gll** [strongswan/strongswan#2484] [General]
We have the idea in our organization to introduce ANYCAST for some core services. Our topology consists of a few HQs which host these core services, and several regional offices which are supposed to communicate with them. 
The regional offices are connected with the HQs via strongswan IPSEC and each regional office has a tunnel for each HQ. 
Using ANYCAST would have the advantage that each regional office would automatically find its nearest HQ and if a HQ would go down it would seamlessly switch to the next HQ for the services.

Now my first thought hearing about the ANYCAST idea was: it wont work, because it would imply that the same SAs would be necessary on each of the tunnels to the HQs. Since the SAs are used to choose which tunnel to take, this would either not be possible to configure or it would lead to traffic arbitrarily being distributed to all HQs.

Now, there is also route-based VPN, and I heard about people even using BGP on their IPSEC-Gateways to distribute traffic to the tunnels. Such a setup would take as much closer to being able to use ANYCAST.

But there still is the big question: would it be possible to use the same SA (or maybe in better wording the same local and remote TS) for several tunnels on the same IPSEC-GW ? Maybe this question is easy to answer and it would help us a lot to decide if we should further pursue the ANYCAST idea.

**Comment by misch42** [comment]
I have set up a route-based VPN with strongswan to enable dynamic routing. Works good.

I think SA sync between the VPN gateways is not necessary. Every gateway does care for its own SAs. If BGP updates the routes, this should update the SAs.

I don't know how this would work with anycast, but this would be interesting.

**Reply by p088gll** [reply]
Thanks Michael, good to know that dynamic routing is working.

**But: my question was about several tunnels with identical local and remote TS on a _single_ strongswan gateway.**

**Reply by misch42** [reply]
Off topic: You need a routing daemon that is able to route via an interface and not only via a gateway. bird is able to do so.  I documented the setup in our blog, but the articles disappeared somehow. Contact me if you need a copy.
